python/main.cc

```diff
@@ -1,106 +0,0 @@
-/* Copyright 2023 CMU, Facebook, LANL, MIT, NVIDIA, and Stanford (alphabetical)
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-#include "legion.h"
-#include "realm/python/python_module.h"
-#include "realm/python/python_source.h"
-
-#include <libgen.h>
-
-#include "flexflow/mapper.h"
-#include "flexflow_c.h"
-using namespace Legion;
-using namespace FlexFlow;
-
-// enum MainTaskIDs {
-//   PYTHON_TOP_LEVEL_TASK_ID = 11111,
-// };
-
-VariantID preregister_python_task_variant(TaskVariantRegistrar const &registrar,
-                                          char const *module_name,
-                                          char const *function_name,
-                                          void const *userdata = NULL,
-                                          size_t userlen = 0) {
-  CodeDescriptor code_desc(
-      Realm::Type::from_cpp_type<Processor::TaskFuncPtr>());
-  code_desc.add_implementation(
-      new Realm::PythonSourceImplementation(module_name, function_name));
-
-  return Runtime::preregister_task_variant(
-      registrar, code_desc, userdata, userlen, registrar.task_variant_name);
-}
-
-int main(int argc, char **argv) {
-#ifdef BINDINGS_AUGMENT_PYTHONPATH
-  // Add the binary directory to PYTHONPATH. This is needed for
-  // in-place builds to find legion.py.
-
-  // Do this before any threads are spawned.
-  {
-    char *bin_path = strdup(argv[0]);
-    assert(bin_path != NULL);
-    char *bin_dir = dirname(bin_path);
-
-    char *previous_python_path = getenv("PYTHONPATH");
-    if (previous_python_path != 0) {
-      size_t bufsize = strlen(previous_python_path) + strlen(bin_dir) + 2;
-      char *buffer = (char *)calloc(bufsize, sizeof(char));
-      assert(buffer != 0);
-
-      // assert(strlen(previous_python_path) + strlen(bin_dir) + 2 < bufsize);
-      // Concatenate bin_dir to the end of PYTHONPATH.
-      bufsize--;
-      strncat(buffer, previous_python_path, bufsize);
-      bufsize -= strlen(previous_python_path);
-      strncat(buffer, ":", bufsize);
-      bufsize -= strlen(":");
-      strncat(buffer, bin_dir, bufsize);
-      bufsize -= strlen(bin_dir);
-      setenv("PYTHONPATH", buffer, true /*overwrite*/);
-    } else {
-      setenv("PYTHONPATH", bin_dir, true /*overwrite*/);
-    }
-
-    free(bin_path);
-  }
-#endif
-
-#ifdef FF_USE_NCCL
```

Comment:
@eddy16112 @jiazhihao do you think the lack of this block of code in the new interface might be related to the nccl-related segfault we're seeing?

Reply:
https://flexflowgroup.slack.com/archives/C046R49TYQK/p1677192959436439

Reply:
We do have the nccl code here https://github.com/flexflow/FlexFlow/blob/python_refactor/src/c/flexflow_c.cc#L1920, not sure if it is too late. One possible solution to try it to move the nccl code before https://gitlab.com/StanfordLegion/legion/-/blob/master/bindings/python/main.cc#L137 
or put `env["NCCL_LAUNCH_MODE"] = "PARALLEL"` before https://github.com/flexflow/FlexFlow/blob/python_refactor/python/flexflow/core/__init__.py#L25